### DISCUSSION TITLE
x_mode on drones

### DISCUSSION THREAD

**Discussion by stonkens** [bitcraze/discussions#1297] [Q&A]
Hi, we're working on getting the cmdVel (in attitude, not attitude rate) working and we were wondering whether +-mode is indeed the default mode as the `commander.py` software indicates. 
We previously worked with older crazyflie software where x-mode (as defined in typical settings, axes 45 degrees offset from propeller axes) was enabled by default, hence were confused what is meant in the cflib software with xmode

Additionally as pitch is negated after the xmode transform we are unsure whether we need to make any adjustments? Would it be possible to generate a figure for the xmode and +-mode configuration and what the values are that it expects (when calling set_setpoint)

Would I be right to assume that if I set xmode to enabled it will follow this coordinate system: https://www.bitcraze.io/documentation/system/platform/cf2-coordinate-system/

**Comment by stonkens** [comment]
We did extensive experiments today to answer these questions:

For our setup (cf 23.11 firmware), when not setting xmode (hence xmode defaulting to false) we have "x-configuration" as indicated https://www.bitcraze.io/documentation/system/platform/cf2-coordinate-system/

I can do a pull request to change the naming convention of x-configuration to +-configuration if desired, lmk

**Reply by knmcguire** [reply]
ah yes please do! standard is indeed the x configuration. Post here the PR if you don't get a response of us quickly to merge it
